_posts/2016-07-07-better-email-validation-with-obey-and-mailgun.markdown

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "Better Email Validation with Obey and Mailgun"
+date: 2016-07-07 10:00:00
+categories: asynchronous validation nodejs
+author: kent.safranski
+cover: /assets/images/covers/books.jpg
+description: Email validation seems simple, but you really should be using a service to accurately validate addresses. Here's how Obey and Mailgun can work together to reliably validate emails.
+comments: true
+---
+
+Validating email is simple, right? Just throw a regex at it? Well it's not as simple as you might think. The RFC Spec (Page 27 of [this monster](https://www.ietf.org/rfc/rfc0822.txt), if you're interested) is clear as mud on what constitutes a valid email.
+
+Basically, with all the special characters and conditions of what can go before and after the `@`, a regex to validate emails accurately would be incomprehensable at best.
+
+_If you'd like to find out more about why email validation sucks check out [this article](http://www.kbedell.com/2011/03/16/how-to-validate-an-email-address-using-regular-expressions/), for now let's move along._
+
+Surely this is something other people have dealt with though, what about a library? Well, they [do exist](https://www.npmjs.com/package/isemail), however, we want to _really_ validate an email. This is most likely either a core communication or authentication property. Libraries can do more than a regex, but they lack more advanced checks against actual resources. Testing for things like not only the validity of a domain, but does it even exist.
```

Comment:
>but they lack more advanced checks

I'd change to

>many lack more advanced checks

Since some _will_ do all of those checks right out from your main app process.

I'd also consider rewording...

>Testing for things like not only the validity of a domain, but does it even exist.

To something like

>For example, a domain may be syntactically valid, but does it have DNS and an MX record? Is it responding to SMTP requests? Can it identify the email user?